### PR TITLE
New grid

### PR DESCRIPTION
**Pull request by xwatkins**
## Purpose
Implement the new grid as discussed.

## Approach
Implement the new grid including the "bleed" class using shadow. This has been tested with uniprot-website.

## Testing
No additional tests.

## Stories
- Updated the documentation of 'Grid'
- Added hooks to Hero Container, and migrated the component to TS

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why
- [x] Were all the edited/created files in TypeScript? If not, please list the files you left in JavaScript and the reason to not update them.
- [x] For the stories you created/updated, are the props modifiables through knobs?

### CODE REVIEW

File: src/styles/components/header.scss [hunk 1]
```diff
@@ -111,27 +111,29 @@ $content-width: 90%;
...
   @include box-shadow();
+  flex-direction: column;
   height: 60vh;
   margin-bottom: 2em;
-  display: grid;
 
-  &__title {
-    grid-column: 2/14;
-    text-align: center;
+  & > * {
```
Comment (xwatkins) [inline src/styles/components/header.scss:118]: Actually this might not be needed...will remove is so

File: stories/Grid.stories.jsx [hunk 2]
```diff
@@ -25,149 +21,81 @@ const gridStyles = {
...
-        span-3
-      </div>
-      <div
-        style={gridStyles.cell}
-        className="uniprot-grid-cell uniprot-grid-cell--span-3"
-      >
-        span-3
-      </div>
+    <div>
+      <h3>Grid modifiers</h3>
```
Comment (dlrice) [inline stories/Grid.stories.jsx:89]: Is it possible to have these as knobs or separate story pages?

File: stories/Grid.stories.jsx [hunk 3]
```diff
@@ -25,149 +21,81 @@ const gridStyles = {
...
+      <p>
+        <code>--centered</code>: To use if the grid should be centered on the
+        page instead of full width.
+      </p>
+      <p>
+        <code>--with-bleed</code>: Sometimes, you need the grid background to
+        overflow into the most left and right gutters. Define the colour by
```
Comment (dlrice) [inline stories/Grid.stories.jsx:96]: ****
```suggestion
        overflow into the leftmost and rightmost gutters. Define the colour by
```

File: src/styles/components/header.scss (hunk 1 above)
Comment (dlrice) [reply src/styles/components/header.scss:118]: Without this the input box shrank
Screenshot 2020-12-21 at 13 56 33

**Review by dlrice — APPROVED** [review]

**Pull request merged**